addons/rmsmartshape/shapes/index_map.gd

```diff
@@ -239,11 +239,13 @@ static func indicies_to_edges(p_indicies: Array[int]) -> Array:
 	return edges
 
 
+## Returns a Dict[Variant, Array[SS2D_IndexMap]]
 static func index_map_array_sort_by_object(imaps: Array) -> Dictionary:
 	var dict := {}
-	for imap in imaps:
+	for imap: SS2D_IndexMap in imaps:
 		if not dict.has(imap.object):
-			var arr: Array[SS2D_IndexMap] = []
-			dict[imap.object] = arr
-		dict[imap.object].push_back(imap)
+			dict[imap.object] = [ imap ]
+		else:
+			var arr: Array[SS2D_IndexMap] = dict[imap.object]
```

Comment:
The static type hint seems to be causing issues with my project's existing shapes. I've been using the official 3.0.0 version from the Godot add-ons. However, when attempting to update to the master version today, I ran into problems with the type hint. Switching from "Array[SS2D_IndexMap]" to just "Array" resolved the issue for me. I haven't had the chance to thoroughly investigate whether it's a problem with Godot's type inference capabilities regarding dictionary members or if it's the wrong type being used here.
![image](https://github.com/SirRamEsq/SmartShape2D/assets/47268404/61c33dd7-ba4e-4899-88e9-04ad46840f12)

Reply:
Yeah, I had my share of those too while porting. Sad to hear it's still an issue.

Reply:
Created an issue #150 to track this bug (and possible future occurences of the same issue).

Reply:
@araslanix Could you please test #151 if it resolves your issue? Otherwise please provide a reproduction project.

Reply:
Hi, sure thing. I'll be back from work in an hour or two and will test it as soon as I get home.

Reply:
![Screenshot 2024-05-17 132408](https://github.com/SirRamEsq/SmartShape2D/assets/47268404/0620d6f9-39f9-4e7e-84cc-14d1a32291ca)

Reply:
Ok, There's a similar issue, but it only appears on the Android build. The PC version works fine:
![image](https://github.com/SirRamEsq/SmartShape2D/assets/47268404/3b706e2c-4485-48b5-b2ce-fa6a92b45762)
same story. Changing array[int] to array solved the problem. apparently godot doesn't understand the nested type: Array[Array[int]]

Reply:
Please don't use this comment section for discussion. Use #150 instead.
I will look into the new problem.